### DISCUSSION TITLE
Lag with GitHub copilot on VS Code

### DISCUSSION THREAD

**Discussion by No767** [community/community#7889] [Copilot Conversations]
I've been using github copilot on VS code for some time now, and noticed that sometimes it starts to lag. Like vs code will start lagging out when im working on a project, and clicking around vs code starts to get laggy. I've tested it without copilot enabled, and the lag someone went away. For some reason, if I use it on intellij and pycharm, there seems to be no lag whatsoever

For reference, I'll also leave my system specs:

CPU - Intel Xeon E5-2609 V2 @ 2.4 GHz
RAM - 52GB DDR3 ECC 
GPU - Nvidia Quadro P4000
VS Code version: Latest 

Edit: It may be because of Kite, not Copiolt. I could be wrong tho

**Comment by Bonfra04** [comment]
Kinda the same problem, yesterday the extension started filling up my CPU usage, disabling the copilot does bring the CPU usage back to normal so something sketchy must be going on. I don't have any other extension installed so it must be it.

**Reply by derycck** [reply]
the same happened to me. With just 1 python notebook open, vscode consumed 60% of the CPU. (intel i7 - 6gen)
Even after closing the notebook and having no py or ipynb files open, the CPU consumption was very high.
I disabled Github Copilot and restarted VSCode, and went back to running the python notebook. CPU consumption became 1%

I don't know if it's a bug in the version, but if it's the new 'normal', the extension becomes infeasible for me. :'(

**Reply by Krzysztof-Cieslak** [reply]
This is discussed in https://github.com/github/feedback/discussions/8861

**Reply by sophia-lawler** [reply]
Im using a 2021 MacBook M1Pro and my battery was lasting 1 hour, my CPU was at 300% and I realised it was Github Copilot.

**Comment by aayushchugh** [comment]
I am using ubuntu 22.04 and vscode starts lagging when I use github copliot
